x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/SharedBytes.java

```diff
@@ -28,136 +35,141 @@
     final int numRegions;
     final long regionSize;
 
+    // TODO: for systems like Windows without true p-write/read support we should split this up into multiple channels since positional
+    // operations in #IO are not contention-free there (https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6265734)
     private final FileChannel fileChannel;
 
-    @SuppressForbidden(reason = "Use positional writes on purpose")
-    SharedBytes(int numRegions, long regionSize, Path file) throws IOException {
+    private final Path path;
+
+    SharedBytes(int numRegions, long regionSize, Environment environment) throws IOException {
+        super("shared-bytes");
         this.numRegions = numRegions;
         this.regionSize = regionSize;
         final long fileSize = numRegions * regionSize;
+        Path cacheFile = null;
         if (fileSize > 0) {
-            this.fileChannel = FileChannel.open(file, OPEN_OPTIONS);
-            // write one byte at the end of the file to make sure all bytes are allocated
-            fileChannel.write(ByteBuffer.allocate(1), fileSize - 1);
+            for (Path path : environment.dataFiles()) {
+                // TODO: be resilient to this check failing and try next path?
```

Comment:
Ideally we would split this up in some form across the available data paths?

Reply:
I guess so (on the splitting up) but I wonder if we should even continue to start up if a given data path is broken?
I certainly see the potential performance benefit of using multiple disks and it's probably nice to be a little more even in the disk use as well.
But we should do this in the follow-up if we want it once we're using multiple file channels I think to not blow this up too much.